### PR TITLE
[14.0][ADD] mrp_subcontracting_picking

### PR DESCRIPTION
**Pull request by ivs-cetmix**
Adding the mrp_subcontracting_picking module

### CODE REVIEW

**Comment by elvise** [comment]
@isserver1 please double check runboat :)

**Comment by elvise** [comment]
@isserver1 gentle reminder about the runboat :)

**Comment by ivs-cetmix** [comment]
@OCA/manufacturing-maintainers could you check this please?

File: mrp_subcontracting_picking/README.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+========================================
```
Comment (pedrobaeza) [inline mrp_subcontracting_picking/README.rst]: This README has incorrect syntax and it's not complete without description

Comment (ivs-cetmix) [reply mrp_subcontracting_picking/README.rst]: Thank you for mentioning this! Fixed this

File: mrp_subcontracting_picking/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Ooops Subcontracting Picking
```
Comment (pedrobaeza) [inline mrp_subcontracting_picking/readme/DESCRIPTION.rst]: Please don't put company name nor title here.

File: mrp_subcontracting_picking/readme/HISTORY.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+14.0.1.0.0
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (pedrobaeza) [inline mrp_subcontracting_picking/readme/HISTORY.rst]: No need of this file, but if put, the underlining should be of the same length of the contents

File: mrp_subcontracting_picking/readme/DESCRIPTION.rst (hunk 2 above)
Comment (ivs-cetmix) [reply mrp_subcontracting_picking/readme/DESCRIPTION.rst]: Done

File: mrp_subcontracting_picking/readme/HISTORY.rst (hunk 3 above)
Comment (ivs-cetmix) [reply mrp_subcontracting_picking/readme/HISTORY.rst]: Done

File: mrp_subcontracting_picking/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Subcontracting Picking
```
Comment (pedrobaeza) [inline mrp_subcontracting_picking/readme/DESCRIPTION.rst:1]: This title should go out.

**Review by pedrobaeza — COMMENTED** [review]
Reading the description, I consider this module is transfiguring somehow the standard of Odoo, as the idea is that you don't do 2 steps: manufacture and receipt the subcontracted material. You do it both things through the reception picking. That's why the are put as ready in the state.

**Comment by elvise** [comment]
@isserver1 can you rebase ?

**Comment by elvise** [comment]
@geomer198 can you rebase ?

File: mrp_subcontracting_picking/models/mrp_production.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    )
+    def _compute_state(self):
+        super(MrpProduction, self)._compute_state()
+        for prod in self.filtered(lambda prod: prod.state == "done"):
+            pickings = prod.move_finished_ids.mapped(
+                "move_dest_ids.picking_id"
+            ).filtered(lambda p: p.state == "subcontracted")
+            if not pickings:
+                continue
+            for picking in pickings:
```
Comment (ivs-cetmix) [inline mrp_subcontracting_picking/models/mrp_production.py]: Why do we need an iteration here?
[action_assign](https://github.com/odoo/odoo/blob/1761d1de485382a5c5dc2d5f421c315ce34e1a3e/addons/stock/models/stock_picking.py#L750) and [action_confirm](https://github.com/odoo/odoo/blob/1761d1de485382a5c5dc2d5f421c315ce34e1a3e/addons/stock/models/stock_picking.py#L738) both work with recordsets

File: mrp_subcontracting_picking/models/mrp_production.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        "product_qty",
+        "qty_producing",
+    )
+    def _compute_state(self):
+        super(MrpProduction, self)._compute_state()
+        for prod in self.filtered(lambda prod: prod.state == "done"):
+            pickings = prod.move_finished_ids.mapped(
+                "move_dest_ids.picking_id"
+            ).filtered(lambda p: p.state == "subcontracted")
+            if not pickings:
```
Comment (ivs-cetmix) [inline mrp_subcontracting_picking/models/mrp_production.py]: why "if not" if we can use 
`if pickings: `

File: mrp_subcontracting_picking/models/mrp_production.py (hunk 5 above)
Comment (geomer198) [reply mrp_subcontracting_picking/models/mrp_production.py]: Resolved

File: mrp_subcontracting_picking/models/mrp_production.py (hunk 6 above)
Comment (geomer198) [reply mrp_subcontracting_picking/models/mrp_production.py]: Resolved

**Pull request closed without merging**